prog/ai/inference_endpoint_nexus.rb

```diff
@@ -75,7 +75,8 @@ def self.assemble(project_id:, location:, boot_image:, name:, vm_size:, storage_
 
   def before_run
     when_destroy_set? do
-      unless ["destroy", "self_destroy"].include?(strand.label)
+      unless inference_endpoint.destroying_set?
```

Comment:
Maybe we should add `unless_*_set?` methods for semaphores?  The current code looks a bit inconsistent in terms of syntax, with `when_destroy_set? do` for the case where it is set, but `unless *.destroying_set?` in the case where it is not.  In addition to looking similar, this should make mocking in the specs easier.  Anyway, just an idea, up to you.

Reply:
I think we previously discussed the block method for checking the absence of the semaphores. If I remember correctly, @fdr might have some thoughts on it. @fdr, what do you think about the `unless_*_set?` methods?

Or should we run checks with `if inference_endpoint.destroyed_set? && ! inference_endpoint.destroying_set?` both with `subject.*_set? instead of mixing `when_*_set?` and `subject.*_set?`?

Reply:
Actually, all our uses of `when_*_set?` syntax can be replaced with if checks. I don't think we have a case where we don't inline the block, right? Should we remove `when_*_set?` completely? (Not necessarily in this PR)

Reply:
@byucesoy As a first step I added `<SEM>_set?` helper to the base prog